### ISSUE TITLE
Api return 404 when keyword contains point symbol

### ISSUE THREAD

**Issue by insoz** [python-gitlab/python-gitlab#1715]
if the branch/commit keyword contains point symbol `.` , the api will return 404 response

example:
`project.branches.get('stbale-1.x')`
`project.comits.get('stbale-1.x')`

------
In page: gl_objects/projects.rst

**Comment by JohnVillalovos** [comment]
Can you run the following and provide the output?

```
#!/usr/bin/env python3

import gitlab
print("gitlab version:", gitlab.__version__)
print("gitlab location:", gitlab.__file__)
```

**Comment by JohnVillalovos** [comment]
As a note, to help your debugging you can do the following, if you want:

```
import gitlab

gl = gitlab.Gitlab(... your values ...)
gl.enable_debug()
```

**Comment by insoz** [comment]
> 

version: '2.10.1'
file: '/Users/insoz/code/ops-fast/venv/lib/python3.7/site-packages/gitlab/__init__.py'

**Comment by JohnVillalovos** [comment]
Thanks. I have a suspicion it is because of:
https://github.com/python-gitlab/python-gitlab/blob/3225f2cfee740374ef36e5cd6796d2370d0e2344/gitlab/utils.py#L65

But not sure if we can just take that out or not. We really need to move to using `urllib.parse.quote()` for this stuff. Though `requests` does some it self and I haven't dug into exactly what we are responsible for and what `requests` will take care of.

**Comment by JohnVillalovos** [comment]
Related issue: https://github.com/python-gitlab/python-gitlab/issues/1356

**Comment by insoz** [comment]
Thanks! BTW. Do we have some plan to use `httpx` or something `asyncio` tool to asynchronous this lib ?

**Comment by nejch** [comment]
> Thanks! BTW. Do we have some plan to use `httpx` or something `asyncio` tool to asynchronous this lib ?

Yes, `httpx` probably, though it still hasn't reached 1.0 and last I checked was missing some proxy or retry features or so, if I'm not mistaken, can't quite remember.
But there is an old PR here that already uses it if you really need it (likely missing lots of recent features): https://github.com/python-gitlab/python-gitlab/pull/1036. That'll need some major rework to update though once we're ready.

**Comment by nejch** [comment]
@JohnVillalovos wasn't this fixed by https://github.com/python-gitlab/python-gitlab/commit/702e41dd0674e76b292d9ea4f559c86f0a99edfe?

**Comment by JohnVillalovos** [comment]
> @JohnVillalovos wasn't this fixed by [702e41d](https://github.com/python-gitlab/python-gitlab/commit/702e41dd0674e76b292d9ea4f559c86f0a99edfe)? If so we can close it, unless you'd like to add a test for that.

It was. I just added PR #1800 to prove it works
